rust/lance/src/index/vector/ivf/builder.rs

```diff
@@ -88,8 +89,9 @@ pub async fn shuffle_dataset(
                     })
                     .collect::<Result<Vec<_>>>()
             })
-            .await?
+            .await
         })
+        .buffer_unordered(num_cpus::get())
```

Comment:
I'm not certain but I think you will be ok with a single `buffer_unordered` call at the end of the pipeline.  Having two doesn't hurt though.